providers/mlx5/mlx5.c

```diff
@@ -609,16 +610,35 @@ static int mlx5_map_internal_clock(struct mlx5_device *mdev,
 	return 0;
 }
 
+struct ibv_cq_ex *mlx5dv_create_cq_ex(struct ibv_context *context,
+				      struct ibv_cq_init_attr_ex *cq_attr,
+				      struct mlx5dv_cq_init_attr_ex *mlx5_cq_attr)
+{
+	return ibv_vendor_create_cq_ex(context, cq_attr, mlx5_cq_attr);
+}
```

Comment:
This seems really silly. Just add a new provider private rdma_init_cq() that contains the init stuff.

Since this is a performance path the driver should tend to its own locking, thus we don't need the strange wrapper.